### PR TITLE
fixes #10225 - handle candlepin events cleanly

### PR DESCRIPTION
**Pull request by thomasmckay**
Simplified code and properly handle entitlement.deleted events. Specific issues revolved around removing a hypervisor's subscription which would trigger a deletion of guest subscriptions.

### CODE REVIEW

**Comment by ehelms** [comment]
With all these updates in this area, starting to wonder if it might just be easier to go ahead and move this area over to scoped search :)

**Comment by thomasmckay** [comment]
@ehelms I'd definitely agree w/ going to scoped search. These changes are destined for 6.0 as the implications of mismatched subscriptions and subscription status are fairly severe.

**Comment by thomasmckay** [comment]
@ehelms @dustints ack or thoughts?

**Comment by ehelms** [comment]
Untested but code looks fine if @dustints wants to test it.

**Comment by dustints** [comment]
@thomasmckay nice, also looks a lot cleaner than before...going to test

File: app/lib/actions/candlepin/reindex_pool_subscription_handler.rb [hunk 1]
```diff
@@ -41,54 +41,41 @@ def handle(message)
...
-        pool_id = message.content['referenceId']
-        @logger.info "re-indexing pools[#{pool_id}] for entitlement[#{message.content['entityId']}]."
+      def reindex_or_unindex_pool(pool_id)
+        ::Katello::Pool.find_pool(pool_id)
         reindex_pool(pool_id)
+      rescue RestClient::ResourceNotFound
+        unindex_pool(pool_id)
       end
 
       def reindex_pool(pool_id)
```
Comment (dustints) [inline app/lib/actions/candlepin/reindex_pool_subscription_handler.rb]: for `reindex_pool`, `unindex_pool` methods, might be good to have a rescue in each?
or in `ListenOnCandlepinEvents` `on_event` rescue from exceptions there?

**Comment by dustints** [comment]
ACK, will tackle other rescues in a different PR

**Pull request merged**